engine/execution/computation/computer/result_collector.go

```diff
@@ -146,8 +154,19 @@ func (collector *resultCollector) CommitCollection(
 	collectionView state.View,
 ) ExecutionStats {
 
-	collector.viewChan <- collectionView
-	collector.eventsListChan <- collector.result.Events[collectionIndex]
+	select {
+	case collector.committerInputChan <- collectionView:
+		// Do nothing
+	case <-collector.committerDoneChan:
+		// Committer exited (probably due to an error)
+	}
+
+	select {
+	case collector.hasherInputChan <- collector.result.Events[collectionIndex]:
+		// Do nothing
+	case <-collector.hasherDoneChan:
+		// Hasher exited (probably due to an error)
```

Comment:
In this case, if there is error, should we return error instead of the ExecutionStats?